src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs

```diff
@@ -224,6 +223,100 @@ public bool IsCaretOnScreen()
                     && caret.Bottom <= advancedView.ViewportBottom;
             });
 
+        public ClassifiedToken[] GetLightbulbPreviewClassifications(
+            string menuText)
+        {
+            return ExecuteOnActiveView(view =>
+            {
+                var broker = GetComponentModel().GetService<ILightBulbBroker>();
+                var classifierAggregatorService = GetComponentModelService<IViewClassifierAggregatorService>();
+                var primitives = GetComponentModelService<IEditorPrimitivesFactoryService>();
+                return GetLightbulbPreviewClassifications(
+                    menuText,
+                    broker,
+                    view,
+                    classifierAggregatorService,
+                    primitives);
+            });
+        }
+
+        public ClassifiedToken[] GetLightbulbPreviewClassifications(
+                    string menuText,
+                    ILightBulbBroker broker,
+                    IWpfTextView view,
+                    IViewClassifierAggregatorService viewClassifierAggregator,
+                    IEditorPrimitivesFactoryService editorPrimitives)
+        {
+            LightBulbHelper.WaitForLightBulbSession(broker, view);
+
+            var bufferType = view.TextBuffer.ContentType.DisplayName;
+            if (!broker.IsLightBulbSessionActive(view))
+            {
+                throw new Exception(string.Format("No Active Smart Tags in View!  Buffer content type={0}", bufferType));
+            }
+
+            var activeSession = broker.GetSession(view);
+            if (activeSession == null || !activeSession.IsExpanded)
+            {
+                throw new InvalidOperationException(string.Format("No expanded light bulb session found after View.ShowSmartTag.  Buffer content type={0}", bufferType));
+            }
+
+            if (!string.IsNullOrEmpty(menuText))
+            {
+                IEnumerable<SuggestedActionSet> actionSets;
+                if (activeSession.TryGetSuggestedActionSets(out actionSets) != QuerySuggestedActionCompletionStatus.Completed)
+                {
+                    actionSets = Array.Empty<SuggestedActionSet>();
+                }
+
+                var set = actionSets.SelectMany(s => s.Actions).FirstOrDefault(a => a.DisplayText == menuText);
+                if (set == null)
+                {
+                    throw new InvalidOperationException(
+                        string.Format("ISuggestionAction {0} not found.  Buffer content type={1}", menuText, bufferType));
+                }
+
+                IWpfTextView preview = null;
+                object pane = HostWaitHelper.PumpingWaitResult(set.GetPreviewAsync(CancellationToken.None));
+                if (pane is System.Windows.Controls.UserControl)
+                {
+                    var container = (pane as System.Windows.Controls.UserControl).FindName("PreviewDockPanel") as DockPanel;
```

Comment:
Or even better, `if (pane is System.Windows.Controls.UserControl control) { var container = control.FindName...`